src/main/java/org/opentripplanner/middleware/models/ItineraryExistence.java

```diff
@@ -95,6 +97,41 @@ public ItineraryExistenceResult getResultForDayOfWeek(DayOfWeek dayOfWeek) {
         throw new IllegalArgumentException("Invalid day of week provided!");
     }
 
+    /**
+     * Helper function to adapt OTP request parameters to GraphQL's JSON {@code variables} object, limited to
+     * variables specified in the GraphQL plan.  Parameter values are all simply Strings, so we can't use
+     * {@code ObjectMapper} to get the types right, must hardcode them specifically for ItineraryExistence so
+     * that the values that are actually strings have {@code \"} around them.
+     */
+    private static String paramsToVariables(Map<String, String> params) {
+        StringBuilder builder = new StringBuilder("{");
```

Comment:
The keys and values we're reading are stored in a `Map<String, String>` so `JsonUtils.toJson` would put quotes around all values.  The values for things like `"arriveBy"` and `"numItineraries"` are not `String`s, so they need the special case to write them without the quotes.  For example it would create `"arriveBy":"false"` rather than `"arriveBy":false`.

I tried a few different approaches to this but they were all more cumbersome than the `StringBuilder` approach.

Reply:
Okay, thanks for the explanation, can you add that in the method's JavaDoc?

Reply:
@JymDyerIBI Would `new Gson().toJson(...)` work for converting these cases to JSON?

Reply:
I've edited the JavaDoc to hopefully be clearer.

I did try `Gson` to create the `"variables"` object. This involves creating a dummy class kept in sync with the GraphQL file, and some of what this helper method did was apply a "different names serializer" for things like changing `mode` into `"modes"`. However it got quite complicated for `"modes"` because it involved `Gson` type adapter features that I couldn't get working and the `Gson` documentation says they discourage the use of.

I also first tried a few GraphQL-specific packages that I hoped would produce a dummy class from `"variables"` but it seems that they all fall short with either variables or mutations.

Reply:
Okay, don't worry about `Gson` now.